vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/test/java/com/vaadin/flow/component/radiobutton/RadioButtonGroupTest.java

```diff
@@ -156,6 +156,28 @@ public void selectDisabledItem_noRedundantEvent() {
         Assert.assertEquals("enabled", event.getValue());
     }
 
+    @Test
+    public void disabledItems_itemEnabledProvider_stayDisabled() {
+        RadioButtonGroup<String> group = new RadioButtonGroup<>();
+        group.setItems("enabled", "disabled");
+        group.setItemEnabledProvider("enabled"::equals);
+
+        List<RadioButton<String>> children = group.getChildren()
+                .map(child -> (RadioButton<String>) child)
+                .collect(Collectors.toList());
+
+        Assert.assertTrue(children.get(0).isEnabled());
+        Assert.assertFalse(children.get(1).isEnabled());
+
+        group.setEnabled(false);
+        Assert.assertFalse(children.get(0).isEnabled());
+        Assert.assertFalse(children.get(1).isEnabled());
```

Comment:
Can you add this block of test again at the end of this method? Just to make sure that if it was disabling again is working.

Reply:
Done.